format/midi/metaevents.go

```diff
@@ -0,0 +1,212 @@
+package midi
+
+import (
+	"github.com/wader/fq/pkg/decode"
+	"github.com/wader/fq/pkg/scalar"
+)
+
+const (
+	SequenceNumber         uint64 = 0x00
+	Text                   uint64 = 0x01
+	Copyright              uint64 = 0x02
+	TrackName              uint64 = 0x03
+	InstrumentName         uint64 = 0x04
+	Lyric                  uint64 = 0x05
+	Marker                 uint64 = 0x06
+	CuePoint               uint64 = 0x07
+	ProgramName            uint64 = 0x08
+	DeviceName             uint64 = 0x09
+	MIDIChannelPrefix      uint64 = 0x20
+	MIDIPort               uint64 = 0x21
+	Tempo                  uint64 = 0x51
+	SMPTEOffset            uint64 = 0x54
+	TimeSignature          uint64 = 0x58
+	KeySignature           uint64 = 0x59
+	EndOfTrack             uint64 = 0x2f
+	SequencerSpecificEvent uint64 = 0x7f
+)
+
+var metaevents = scalar.UintMapSymStr{
+	SequenceNumber:         "sequence_number",
+	Text:                   "text",
+	Copyright:              "copyright",
+	TrackName:              "track_name",
+	InstrumentName:         "instrument_name",
+	Lyric:                  "lyric",
+	Marker:                 "marker",
+	CuePoint:               "cue_point",
+	ProgramName:            "program_name",
+	DeviceName:             "device_name",
+	MIDIChannelPrefix:      "midi_channel_prefix",
+	MIDIPort:               "midi_port",
+	Tempo:                  "tempo",
+	SMPTEOffset:            "smpte_offset",
+	TimeSignature:          "time_signature",
+	KeySignature:           "key_signature",
+	EndOfTrack:             "end_of_track",
+	SequencerSpecificEvent: "sequencer_specific_event",
+}
+
+var metafns = map[uint64]func(d *decode.D){
+	SequenceNumber:         decodeSequenceNumber,
+	Text:                   decodeText,
+	Copyright:              decodeCopyright,
+	TrackName:              decodeTrackName,
+	InstrumentName:         decodeInstrumentName,
+	Lyric:                  decodeLyric,
+	Marker:                 decodeMarker,
+	CuePoint:               decodeCuePoint,
+	ProgramName:            decodeProgramName,
+	DeviceName:             decodeDeviceName,
+	MIDIChannelPrefix:      decodeMIDIChannelPrefix,
+	MIDIPort:               decodeMIDIPort,
+	Tempo:                  decodeTempo,
+	SMPTEOffset:            decodeSMPTEOffset,
+	TimeSignature:          decodeTimeSignature,
+	KeySignature:           decodeKeySignature,
+	EndOfTrack:             decodeEndOfTrack,
+	SequencerSpecificEvent: decodeSequencerSpecificEvent,
+}
+
+func decodeMetaEvent(d *decode.D, event uint8, ctx *context) {
+	ctx.running = 0x00
+	ctx.casio = false
+
+	delta := func(d *decode.D) {
+		ctx.tick += d.FieldUintFn("delta", vlq)
+		d.FieldValueUint("tick", ctx.tick)
+	}
+
+	if fn, ok := metafns[uint64(event)]; ok {
+		d.FieldStruct("metaevent", func(d *decode.D) {
```

Comment:
known as "metaevent" or can be meta_event?

Reply:
There's nothing super-official .. after a brief survey of the Internet it seems the most common usage is either _Meta Event_ or _Meta-Event_ with _Metaevent_ coming in a poor third, so _snake case_ it is :-).

Fixed in [87c80f5](https://github.com/wader/fq/pull/1004/commits/87c80f57761bd705b39edae8ee9457cd9b951db6)

Reply:
👍 makes sense